src/bundles/arcade_2d/audio.ts

```diff
@@ -35,43 +35,47 @@ export class AudioClip {
     if (url === '') {
       throw new Error('AudioClip URL cannot be empty');
     }
-    if (AudioClip.audioClips.has(url)) {
-      return AudioClip.audioClipsArray[AudioClip.audioClips.get(url) as number];
+    if (AudioClip.audioClipsIndexMap.has(url)) {
+      return AudioClip.audioClipsArray[AudioClip.audioClipsIndexMap.get(url) as number];
     }
     return new AudioClip(url, volumeLevel);
   }
   public getUrl() {
     return this.url;
   }
-  public getVolume() {
+  public getVolumeLevel() {
     return this.volumeLevel;
   }
-  public getLoop() {
-    return this.loop;
+  public shouldAudioClipLoop() {
+    return this.shouldLoop;
   }
-  public shouldPlayClip() {
-    return this.playClip;
+  public shouldAudioClipPlay() {
+    return this.shouldPlay;
   }
-  public loopAudioClip(loop: boolean) {
-    if (this.loop !== loop) {
-      this.loop = loop;
-      this.audioClipNotUpdated = true;
+  public setShouldAudioClipLoop(loop: boolean) {
+    if (this.shouldLoop !== loop) {
+      this.shouldLoop = loop;
+      this.isUpdated = false;
     }
   }
-  public playAudioClip(play: boolean) {
-    this.playClip = play;
-    this.audioClipNotUpdated = true;
+  /**
+   * Updates the play/pause state.
+   * @param play When true, the Audio Clip has a playing state.
+   */
+  public setShouldAudioClipPlay(play: boolean) {
+    this.shouldPlay = play;
+    this.isUpdated = false;
   }
   /**
    * Checks if the Audio Clip needs to update. Updates the flag if true.
    */
   public hasAudioClipUpdates() {
-    const temp = this.audioClipNotUpdated;
-    this.updatedAudioClip();
+    const temp = !this.isUpdated;
+    this.setAudioClipUpdated();
     return temp;

```

Comment:
Minor nit. Since this is not exactly a "temporary" variable (the value is returned, after all), perhaps it would make more sense to call this `prevValue`?

```suggestion
    const prevValue = !this.isUpdated;
    this.setAudioClipUpdated();
    return prevValue;
```